### PR TITLE
lib-user: Add gradient to BarChart bars

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Linked Issue and/or Talk Post
- not exactly, but closer to [related Figma design](https://www.figma.com/design/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?node-id=96-831&m=dev)

## Describe your changes
- add bar color gradient to BarChart

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR? reviewing stories probably sufficient
- Which storybook stories should be reviewed?
  - http://localhost:6008/?path=/story/components-shared-barchart--last-7-days
- Are there plans for follow up PR’s to further fix this bug or develop this feature? see linked GitHub project

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by mcbouslog** [comment]
@seanmiller26 - this adds a bar color gradient to BarChart as follows. It's not exactly like in the Figma design, where each bar goes from the same light to dark teal. Instead, with these changes, the gradient goes from the max value of the data to 0. We can revisit exactly matching Figma later, I'm not sure that's doable with Grommet's DataChart (what BarChart uses), but I think this is a good step in the right direction.

The way the Grommet DataChart bar color gradient works is that it takes two specific colors and creates a gradient between them. So these changes reflect a gradient between "brand" and "neutral-1." I don't think I can do a single linear-gradient like we might normally do, with css, but I have to use two colors as noted. Accordingly, would you prefer a slightly lighter color than "brand"? If so, let me know. Thanks!

BarChart_with_color_gradient

**Comment by seanmiller26** [comment]
This looks really good! No need to match Figma, this'll do nicely.

File: packages/lib-user/src/components/shared/BarChart/BarChart.js [hunk 1]
```diff
@@ -18,19 +18,34 @@ function BarChart({
...
+  const max = Math.max(...types)
+  const gradient = [
+    { value: 0, color: 'neutral-1' },
+    { value: max, color: 'brand' }
+  ]
 
   // set chart options based on screen size and data length
   const chartOptions = {
-    color: 'brand',
+    color: gradient,
```
Comment (mcbouslog) [inline packages/lib-user/src/components/shared/BarChart/BarChart.js:48]: Related Grommet Chart storybook - https://storybook.grommet.io/?path=/story/visualizations-chart-gradient--gradient-charts

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: lib-user

This PR updates the design of the BarCharts (used on the user stats page, e.g. app-root's local.zooniverse.org:3000/users/darkeshard/stats). The bars now have a subtle colour gradient.

LGTM 👍

**Pull request merged**